### ISSUE TITLE
[REVIEW]: SimplyAgree: An R package and jamovi Module for Simplifying Agreement and Reliability Analyses

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#4148]
**Submitting author:** @arcaldwell49 (<a href="http://orcid.org/0000-0002-4541-6283">Aaron R. Caldwell</a>)
**Repository:** https://github.com/arcaldwell49/SimplyAgree
**Branch with paper.md** (empty if default branch): 
**Version:** v0.0.3
**Editor:** @fboehm
**Reviewers:** @slwu89, @itismeghasyam
**Archive:** 10.5281/zenodo.6338611

 

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/576840a8539ab55e78ef32a233570bfd/status.svg)](https://joss.theoj.org/papers/576840a8539ab55e78ef32a233570bfd)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/576840a8539ab55e78ef32a233570bfd"><img src="https://joss.theoj.org/papers/576840a8539ab55e78ef32a233570bfd/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/576840a8539ab55e78ef32a233570bfd/status.svg)](https://joss.theoj.org/papers/576840a8539ab55e78ef32a233570bfd)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/arcaldwell49/SimplyAgree" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@slwu89 & @itismeghasyam, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @fboehm know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @slwu89
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/arcaldwell49/SimplyAgree">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/arcaldwell49/SimplyAgree">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@arcaldwell49) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @itismeghasyam
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/arcaldwell49/SimplyAgree">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/arcaldwell49/SimplyAgree">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@arcaldwell49) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] #4214
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.65 s (689.7 files/s, 163010.9 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
HTML                           162           4584            728          55787
R                              116           1745           1999           6351
CSS                             10           1103             68           6151
Markdown                        35           2837              0           5533
Rmd                             78           4381           7879           3781
XML                              4              8              3            751
YAML                            15             69             11            714
TeX                              4             68              9            643
JavaScript                      12            102             53            601
C/C++ Header                     1             20              0             60
SVG                             11              0              1             58
Bourne Shell                     1              5              0             13
-------------------------------------------------------------------------------
SUM:                           449          14922          10751          80443
-------------------------------------------------------------------------------


Statistical information for the repository 'c7969f9dd5b62b9388919608' was
gathered on 2022/02/10.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Aaron Caldwell                  10          1343            144           79.48
arcaldwell49                     2            24            360           20.52

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
arcaldwell49                863         3595.8          0.3                6.72
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1080/19466315.2019.1677495 is OK
- 10.1016/s0140-6736(86)90837-8 is OK
- 10.1177/096228029900800204 is OK
- 10.2307/2532051 is OK
- 10.1177/0962280211402548 is OK
- 10.1002/sim.2778 is OK
- 10.1080/10543400802527890 is OK
- 10.1016/j.cmpb.2012.09.002 is OK
- 10.1515/ijb-2015-0039 is OK

MISSING DOIs

- 10.1519/15184.1 may be a valid DOI for title: Quantifying Test-Retest Reliability Using the Intraclass Correlation Coefficient and the SEM

INVALID DOIs

- None
```

**Comment by fboehm** [comment]
@itismeghasyam and @slwu89 - how are your reviews going? Is there anything that I can assist with? Do you have any questions? Thanks again!

**Comment by slwu89** [comment]
@whedon generate pdf

**Comment by slwu89** [comment]
Hi @fboehm, how do I regenerate the PDF? It looks like the old whedon command doesn't work. Thanks!

**Comment by slwu89** [comment]
@editorialbot generate pdf

**Comment by slwu89** [comment]
I am linking issues created in the review repo to this review here:

- community guidelines https://github.com/arcaldwell49/SimplyAgree/issues/10
- finding documentation https://github.com/arcaldwell49/SimplyAgree/issues/9

**Comment by slwu89** [comment]
@fboehm I have a question, in the documentation section of the review checking it indicates:

> A statement of need: Do the authors clearly state what problems the software is designed to solve and who the target audience is?

Does this mean that the README file needs to have a section titled "Statement of need" or is it all right if this information is available but not labeled as such?

**Comment by slwu89** [comment]
@editorialbot generate pdf

**Comment by slwu89** [comment]
@fboehm I've completed my review, I recommend accepting for JOSS. Thanks.

**Comment by itismeghasyam** [comment]
Well written paper and well documented package. I have only one issue with Installation instructions: dependencies. I'm linking it here [https://github.com/openjournals/joss-reviews/issues/4214](https://github.com/openjournals/joss-reviews/issues/4214)

**Comment by arcaldwell49** [comment]
@itismeghasyam  `cccrm` and `sjstats` have been removed as required dependencies for v0.0.3 now on the main branch.

`cccrm` is only suggested for the purposes of the vignettes.

**Comment by fboehm** [comment]
@itismeghasyam - are you satisfied with the correction to fix the one small issue above? If so, then we can proceed towards publication. Thanks again!

**Comment by itismeghasyam** [comment]
@fboehm Yes, looks good to me!

**Comment by fboehm** [comment]
Thanks, @itismeghasyam !

**Comment by fboehm** [comment]
@editorialbot generate pdf

**Comment by fboehm** [comment]
@arcaldwell49 - the reviewers have recommended your submission for publication. I'll proofread the pdf in the next couple of days. Once that is completed, a few minor fixes may be needed. Eventually, I'll ask you to release a new version of the package and to make an archive, but please wait on those tasks until I instruct you to do them. Thanks again!

**Comment by fboehm** [comment]
@arcaldwell49 - the pdf paper looks really good! please make the minor corrections below.

A few minor fixes are needed in the pdf. I think that these can be accomplished by updating the bib file as needed.

line 110: R needs to be capitalized. You can accomplish this by putting braces around capital R in your bib file, like {R}
line 115: capitalize R.
line 117: Do you need to capitalize Bland-Altman? I'm guessing you do.
line 122: Capitalize Bland-Altman
line 139: Capitalize BlandAltman

Once these fixes are made, we can proceed towards publication.

**Comment by arcaldwell49** [comment]
@fboehm good catches on my dumb typos!

All requested typos fixed.

**Comment by fboehm** [comment]
Thank you, @arcaldwell49 !

**Comment by fboehm** [comment]
@editorialbot generate pdf

**Comment by fboehm** [comment]
@editorialbot check references

**Comment by fboehm** [comment]
@arcaldwell49 - can you double-check the DOIs that are mentioned in the above comment? If the editorialbot gave the correct doi for the "Quantifying Test-Retest..." article, please add it to the bib file so that it appears in the pdf. Also, please update the doi that is supposedly invalid. Thanks again!

**Comment by arcaldwell49** [comment]
Oh wow, good catch! It seems JSCR has resolved the doi issue. Pushing new version in a minute...

**Comment by fboehm** [comment]
@editorialbot check references

**Comment by fboehm** [comment]
@editorialbot generate pdf

**Comment by arcaldwell49** [comment]
Hi @fboehm, I will make the tagged release ASAP. As far as the archive goes, are you wanting something like a Zenodo archive of the paper.md file?

**Comment by arcaldwell49** [comment]
@fboehm Release is up. https://github.com/arcaldwell49/SimplyAgree/releases/tag/0.0.3

 I have a draft in Zenodo of v0.0.3 that I will publish with your confirmation that this is how you would like me to proceed.

**Comment by fboehm** [comment]
Thanks, @arcaldwell49 . The archive should contain the entire repository, not just the paper.md file. I think that what you've shared is what's needed. Please correct me if I'm mistaken.

**Comment by fboehm** [comment]
Please publish it and report the doi here.

**Comment by fboehm** [comment]
Thanks, @arcaldwell49 !

**Comment by fboehm** [comment]
@editorialbot set v0.0.3 as version

**Comment by fboehm** [comment]
@editorialbot set 10.5281/zenodo.6338611 as archive

**Comment by fboehm** [comment]
@editorialbot check references

**Comment by fboehm** [comment]
@editorialbot recommend-accept

**Comment by Kevin-Mattheus-Moerman** [comment]
@arcaldwell49 
- [ ] Please add city and country information to your affiliations, for consistency with other submissions. 
- [ ]

**Comment by Kevin-Mattheus-Moerman** [comment]
@openjournals/dev @xuanxu @arfon I think we have an issue with multiple affiliations for one author. Here the paper features the yaml `affiliation: "1, 2"` However, the affiliation appears as superscript `12` rather than `1, 2`. 


![Screenshot from 2022-03-12 08-38-14](https://user-images.githubusercontent.com/8392709/158010764-81f0f8c5-6eb1-4fd5-bda7-d07f3347390e.png)

**Comment by Kevin-Mattheus-Moerman** [comment]
@arcaldwell49 can you check the comments above :point_up: ?

**Comment by Kevin-Mattheus-Moerman** [comment]
@openjournals/dev can you check my query?

**Comment by danielskatz** [comment]
Yes, looks like a bug - @xuanxu ?

**Comment by arcaldwell49** [comment]
@Kevin-Mattheus-Moerman  Apologies, I thought I posted this here, but the paper has been updated to address those comments

https://raw.githubusercontent.com/arcaldwell49/SimplyAgree/master/paper/paper.md

**Comment by xuanxu** [comment]
> I think we have an issue with multiple affiliations for one author. Here the paper features the yaml `affiliation: "1, 2"` However, the affiliation appears as superscript `12` rather than `1, 2`.

Fix deployed. This should be `1, 2` now.

**Comment by danielskatz** [comment]
@editorialbot recommend-accept

to verify @xuanxu's fix message

**Comment by danielskatz** [comment]
Thanks @xuanxu - looks good.

**Comment by danielskatz** [comment]
👋 @Kevin-Mattheus-Moerman - back to you, or let me know if you want me to wrap things up since I'm on duty this week

**Comment by Kevin-Mattheus-Moerman** [comment]
@arcaldwell49 thanks for making those changes. And thank you @xuanxu for fixing that bug. We are good to proceed now.

**Comment by Kevin-Mattheus-Moerman** [comment]
@arcaldwell49 congratulations on getting this work published in JOSS! 

A special thank you to @slwu89 and @itismeghasyam for reviewing this work!

Also, thanks @fboehm for editing this one.
